spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2002,13 +2002,17 @@ def __mark_occurrences(self):
         cursor = self.__find_first(text)
         self.occurrences = []
         extra_selections = self.get_extra_selections('occurrences')
-
+        first_ocurrence = None
         while cursor:
             self.occurrences.append(cursor.blockNumber())
-            selection = self.get_selection(
-                cursor, background_color=self.occurrence_color)
-            if selection and selection.cursor.selectedText():
+            selection = self.get_selection(cursor)
+            if len(selection.cursor.selectedText()) > 0:
                 extra_selections.append(selection)
+                if len(extra_selections) == 1:
+                    first_ocurrence = selection
+                else:
+                    selection.format.setBackground(self.occurrence_color)
+                    first_ocurrence.format.setBackground(self.occurrence_color)

```

Comment:
I don't understand how `first_ocurrence` is set to something diffrent than `None` if the condition `len(extra_selections) == 1` is not `True`.

Could you explain that?

Reply:
The `first_occurrence` variable is necessary because we only want to paint the background where there's at least one more. During this while the file is being seen from the start to the bottom, so we need to save the first occurrence, without painting, so we can paint its background if we find another one in the document.